src/pluto/image.cr

```diff
@@ -2,67 +2,43 @@ require "./bindings/*"
 require "./format/*"
 require "./operation/*"
 
-class Pluto::Image
-  include Format::JPEG
-  include Format::PPM
-
-  include Operation::BilinearResize
-  include Operation::BoxBlur
-  include Operation::Brightness
-  include Operation::ChannelSwap
-  include Operation::Contrast
-  include Operation::GaussianBlur
-  include Operation::HorizontalBlur
-  include Operation::VerticalBlur
-
-  property red : Array(UInt8)
-  property green : Array(UInt8)
-  property blue : Array(UInt8)
-  property alpha : Array(UInt8)
-  property width : Int32
-  property height : Int32
-
-  def initialize(@red, @green, @blue, @alpha, @width, @height)
-  end
-
-  def clone : Image
-    self.class.new(
-      @red.clone,
-      @green.clone,
-      @blue.clone,
-      @alpha.clone,
-      @width,
-      @height
-    )
+abstract class Pluto::Image
+  macro inherited
+    include Format::JPEG
+    include Format::PPM
+    include Format::Save
+    extend Format::Open
+
+    include Operation::BilinearResize
+    include Operation::BoxBlur
+    include Operation::Brightness
+    include Operation::ChannelSwap
+    include Operation::Contrast
+    include Operation::GaussianBlur
+    include Operation::HorizontalBlur
+    include Operation::VerticalBlur
   end
 
-  def each_channel(&) : Nil
-    yield @red, ChannelType::Red
-    yield @green, ChannelType::Green
-    yield @blue, ChannelType::Blue
-    yield @alpha, ChannelType::Alpha
-    nil
+  macro forward_to_rgb_image(*methods)
+    {% for method in methods %}
+      @[Deprecated("Use RGBImage.{{method.id}} instead")]
+      def self.{{method.id}}(*args, **kwargs)
+        RGBImage.{{method.id}}(*args, **kwargs)
+      end
+    {% end %}
   end
 
-  def [](channel_type : ChannelType) : Array(UInt8)
-    case channel_type
-    in ChannelType::Red   then @red
-    in ChannelType::Green then @green
-    in ChannelType::Blue  then @blue
-    in ChannelType::Alpha then @alpha
-    end
-  end
+  forward_to_rgb_image open, from_ppm, from_jpg
```

Comment:
Added this macro for backwards compatibility when changing from `Image` -> `RGBImage`.